hail/python/hail/matrixtable.py

```diff
@@ -2223,7 +2223,7 @@ def aggregate_cols(self, expr, _localize=True) -> Any:
         """
         base, _ = self._process_joins(expr)
         analyze('MatrixTable.aggregate_cols', expr, self._global_indices, {self._col_axis})
-        cols_table = ir.MatrixColsTable(base._mir)
+        cols_table = ir.MatrixColsTable(ir.MatrixMapCols(base._mir, base.col._ir, []))
```

Comment:
Why is it kosher for map cols to drop the key? Shouldn't we need to use a key by for that?

Reply:
don't actually have a MatrixKeyColsBy. MMC does both. We could separate it out, but don't have separate nodes because we (a) don't really care about optimizing around col keys since they're unordered, and (b) since the col key is totally ignored as we lower.

Reply:
Sorry if I'm being dense, but isn't ordering relevant to aggregation? I'm thinking of `hl.agg.collect` in particular. I thought it was true that the results came in key-order. If ordering is important, why is it safe to drop the key before `ir.TableAggregate`?

Reply:
This PR makes a semantic change in aggregation order (currently in sort order aka order of cols(), but desired behavior is  matrixtable column order (localize_entries order). This is what we agreed in team meeting the week before the IBG workshop, but let's revisit Wednesday to make sure we're still on the same page.